### PR TITLE
Complete smtplib and sync with 3.5+

### PR DESCRIPTION
**Pull request by scop**

### CODE REVIEW

File: stdlib/3/smtplib.pyi [hunk 1]
```diff
@@ -39,71 +44,77 @@ class SMTPConnectError(SMTPResponseException): ...
...
     file = ...  # type: Any
-    helo_resp = ...  # type: Any
-    ehlo_msg = ...  # type: Any
-    ehlo_resp = ...  # type: Any
-    does_esmtp = ...  # type: Any
-    default_port = ...  # type: Any
+    helo_resp = ...  # type: Optional[bytes]
+    ehlo_msg = ...  # type: str
+    ehlo_resp = ...  # type: Optional[bytes]
+    does_esmtp = ...  # type: bool
```
Comment (JelleZijlstra) [inline stdlib/3/smtplib.pyi]: Actually an int (although used like a bool).

File: stdlib/3/smtplib.pyi [hunk 2]
```diff
@@ -39,71 +44,77 @@ class SMTPConnectError(SMTPResponseException): ...
...
     def quit(self) -> _Reply: ...
 
 class SMTP_SSL(SMTP):
-    default_port = ...  # type: Any
-    keyfile = ...  # type: Any
-    certfile = ...  # type: Any
-    context = ...  # type: Any
-    def __init__(self, host=..., port=..., local_hostname=..., keyfile=..., certfile=...,
-                 timeout=..., source_address=..., context=...): ...
+    keyfile = ...  # type: Optional[str]
```
Comment (JelleZijlstra) [inline stdlib/3/smtplib.pyi]: We prefer variable annotations (`x: int`) in new code.

File: stdlib/3/smtplib.pyi (hunk 1 above)
Comment (scop) [reply stdlib/3/smtplib.pyi]: I actually have a PR open on that -- it's also documented to be used as a bool.
https://github.com/python/cpython/pull/8718

Comment (scop) [reply stdlib/3/smtplib.pyi]: ...but will change to int in the meantime.

File: stdlib/3/smtplib.pyi (hunk 2 above)
Comment (scop) [reply stdlib/3/smtplib.pyi]: Hm, good to know, but this wasn't really new code -- the particular highlighted line was changed from `keyfile = ...  # type: Any` to `keyfile = ...  # type: Optional[str]`, and similar changes were made to `certfile` and `context`.

There are quite a few variables annotated using the older formatting in smtplib.pyi, would you like me to change all of them to use variable annotations instead?

Comment (JelleZijlstra) [reply stdlib/3/smtplib.pyi]: Yes, let's just change them all. Thanks!

**Pull request merged**